gm4/plugins/annotations.py

```diff
@@ -0,0 +1,68 @@
+from beet import Context
+import logging
+import re
+from functools import partial
+from pathlib import Path
+
+def beet_default(ctx: Context):
+    """Sets up a logging handler to repeat build log entries with the github action annotation format"""
+    root_logger = logging.getLogger(None) # get root logger
+
+    handler = logging.StreamHandler()
+    handler.setFormatter(AnnotationFormatter())
+
+    def filter(record: logging.LogRecord):
+        if record.name == "time":
+            return False # disable annotations for time - is spammy in debug mode
+        if record.name == "gm4":
+            return False # disable just the root gm4 logger, permit all children to log.
+                            # prevents the "found X modules" and "Finished" messages
+        return True
+
+    handler.addFilter(filter)
```

Comment:
You know I kinda forgot we could just remove the `gm4.plugins.list_modules` from the config, though I kinda like the idea of it still showing up in the console output, just not in the github summary.

As for the `time` messages, those show up when you run a beet pipeline in debug mode, timing each subpipeline. I *think* the preferred logger interface is to setup filters blocking your desired messages, rather then deleting that logger directly. That way a second log output (eg to file) would still have all the events.

Reply:
Ah this filter is only for the GH annotations, then I agree removing the log messages in the first place is not a good solution. An alternative is to filter out debug and info log levels, as I suspect we may add more of those in the future and I don't immediately see a use for them as GH annotations.

Reply:
Correct, this filter only applies to the StreamHandler setup by `gm4.plugins.annotations`. 

I kept INFO since we use that for messages like "published update patch to modrinth" which would be nice to see in the summary. 

DEBUG only shows up if we run the action with the "debug logging" flag enabled (it forwards that to `beet -l DEBUG build`, so it would be handy to get a pretty-printed summary of those whenever we debug an action.